code/game/machinery/requests_console.dm

```diff
@@ -102,6 +97,13 @@ GLOBAL_LIST_EMPTY(allRequestConsoles)
 	Radio.follow_target = src
 	. = ..()
 
+	var/area/containing_area = get_area(src)
+	if(isnull(department))
+		department = containing_area.request_console_name || containing_area.name
+	if(isnull(departmentType))
+		departmentType = containing_area.request_console_flags
+	announcementConsole ||= containing_area.request_console_announces
```

Comment:
is this mean to be `|=`?

Reply:
This is supposed to cleverly toggle announcementConsole if either it or the area's setting is true but I can probably just replace it with an assignment

Reply:
byond and its niche operators :O